### PR TITLE
DX: PR friendly, branch-specific PHP CS Fixer cache files

### PR DESCRIPTION
**Pull request by keradus**
appendix to 
 https://github.com/sebastianbergmann/phpunit/commit/59f3dc25d0054cb437293dcf8003d04358b03b8b

sth to consider, as each branch is having configured composer branch alias.

Right now if I have a local branch 9.6 and I run Fixer for it, it creates a dedicated cache file. Then, I create a 9.6-fix-typo branch, which is very similar to 9.6 branch, yet have a separated cache file.

With proposed way, 9.6-fix-typo will reuse 9.6 cache file.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 # PHP-CS-Fixer
 /.php-cs-fixer.php
-/.php-cs-fixer.cache
+/.php-cs-fixer.cache/
```
Comment (sebastianbergmann) [inline .gitignore:17]: This seems to be unrelated.

File: .php-cs-fixer.dist.php [hunk 2]
```diff
@@ -342,6 +342,6 @@
         'whitespace_after_comma_in_array' => true,
     ]);
 
-$config->setCacheFile(__DIR__ . '/.php-cs-fixer.cache/' . sha1(@trim((string) @shell_exec('git rev-parse --abbrev-ref HEAD'))));
+$config->setCacheFile(__DIR__ . '/.php-cs-fixer.cache/' . json_decode((string) @file_get_contents('composer.json'), true)["extra"]["branch-alias"]["dev-master"] ?? 'unknown');
```
Comment (sebastianbergmann) [inline .php-cs-fixer.dist.php:345]: Interesting idea.

File: .gitignore (hunk 1 above)
Comment (keradus) [reply .gitignore:17]: > chore: indicate that entry is dir and not regular file

I was fooled that there is a cache file, till I realised it's actually a dir. And as .gitignore allows to specify that entry is a dir, I use it when applicable.
